### ISSUE TITLE
Incompatible with Pandoc v2

### ISSUE THREAD

**Issue by fny** [rstudio/rmarkdown#1244]
Pandoc v2 was recently released and has several breaking changes. In this case, the pandoc step failed since the `--smarty` option has now been turned into an "extension." 

```
/usr/local/bin/pandoc +RTS -K512m -RTS Example.utf8.md --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output Example.html --smart --email-obfuscation none --self-contained --standalone --section-divs --template /usr/local/lib/R/3.4/site-library/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --variable 'theme:bootstrap' --include-in-header /var/folders/lg/m3q3m8z11458bpm2mbnl235h0000gn/T//Rtmp2mNLGC/rmarkdown-str69ee47fdad46.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' 
--smart/-S has been removed.  Use +smart or -smart extension instead.
For example: pandoc -f markdown+smart -t markdown-smart.
Try pandoc --help for more information.
```

Same business with PDF generation:

```
/usr/local/bin/pandoc +RTS -K512m -RTS Example.utf8.md --to latex --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output Example.pdf --template /usr/local/lib/R/3.4/site-library/rmarkdown/rmd/latex/default-1.17.0.2.tex --highlight-style tango --latex-engine /Library/TeX/texbin/pdflatex --variable graphics=yes --variable 'geometry:margin=1in' 
output file: Example.knit.md

--latex-engine has been removed.  Use --pdf-engine instead.
Try pandoc --help for more information.
Error: pandoc document conversion failed with error 2
Execution halted
```

**Comment by yihui** [comment]
Duplicate of #1120

**Comment by fny** [comment]
I actually did run `update.packages('rmarkdown')` and was running v1.8, which didn't seem to fix anything. Has the pandoc v2 compatible version been released to CRAN?

After reading through the source, I did notice some fixes for v2, so I tried `devtools::install_github("rstudio/rmarkdown")` which worked. Honestly, I should have tried that before reporting a bug, but  I'm thinking it might be a good idea to explicitly address this in the README somewhere? Maybe the PANDOC.md installation doc?

Cheers, and many thanks for your immense contributions to the R ecosystem!
